purchase_discount/models/purchase_order.py

```diff
@@ -25,10 +10,14 @@
 class PurchaseOrderLine(models.Model):
     _inherit = "purchase.order.line"
 
-    @api.model
-    def _calc_line_base_price(self, line):
-        res = super(PurchaseOrderLine, self)._calc_line_base_price(line)
-        return res * (1 - line.discount / 100.0)
+    def _calc_line_base_price(self):
+        self.ensure_one()
+        res = super(PurchaseOrderLine, self)._calc_line_base_price()
+        return res * (1 - self.discount / 100.0)
+
+    @api.depends('discount')
+    def _compute_amount(self):
+        super(PurchaseOrderLine, self)._compute_amount()
```

Comment:
I just ported what @pedrobaeza did and make it works the same as possible like it was into 8.0. I think improvements like this can be add into other PR according a quick talk with the author about why it was not done before